src/components/Banners/Banner/Banner.tsx

```diff
@@ -0,0 +1,155 @@
+//- React Imports
+import React, { useEffect, useState } from 'react';
+import { useHistory } from 'react-router-dom';
+
+//- Component Imports
+import { BannerContent } from 'components';
+
+//- Utils Imports
+import { BannerContentType, getBannerContent } from './utils';
+import { isWilderWorldAppDomain } from 'lib/utils/url';
+
+//- Banner Data
+import BannerData from './BannerData.json';
+
+const Banner: React.FC = () => {
+	//////////////////
+	//     Hooks    //
+	//////////////////
+
+	const history = useHistory();
+
+	//////////////////
+	//     Data     //
+	//////////////////
+
+	const currentTime = new Date().getTime();
+	const contentData: BannerContentType[] = BannerData;
+
+	//////////////////
+	//    State     //
+	//////////////////
+
+	const [backgroundBlob, setBackgroundBlob] = useState<string | undefined>();
+	const [shouldDisplayBanner, setShouldDisplayBanner] = useState<boolean>(true);
+	const [shouldDisplayCountdown, setShouldDisplayCountdown] =
+		useState<boolean>(false);
+	const [bannerContent, setBannerContent] = useState<
+		BannerContentType | undefined
+	>();
+
+	//////////////////
+	//    Effects   //
+	//////////////////
+	console.log('currentTime', currentTime);
+	console.log(currentTime + 3 * 60 * 1000);
+
+	// Set banner
+	useEffect(() => {
+		let isActive = true;
+
+		const content = getBannerContent(currentTime, contentData);
+		setBannerContent(content);
+
+		if (isActive) {
+			console.log('console.log(bannerContent);');
+			if (!bannerContent) {
+				setShouldDisplayBanner(false);
+				console.log('NO DATA');
+			} else {
+				setShouldDisplayBanner(true);
+				if (
+					bannerContent.countdownDate &&
+					currentTime < bannerContent.countdownDate
+				) {
+					setShouldDisplayCountdown(true);
+				}
+			}
+		}
+	}, [bannerContent, contentData, currentTime]);
+
+	// Set background image
+	useEffect(() => {
+		if (!bannerContent?.imgUrl) {
+			return;
+		}
+
+		let isMounted = true;
+
+		fetch(bannerContent.imgUrl)
+			.then((r) => r.blob())
+			.then((blob) => {
+				if (isMounted) {
+					const url = URL.createObjectURL(blob);
+					setBackgroundBlob(url);
+				}
+			});
+		return () => {
+			isMounted = false;
+		};
+	}, [bannerContent?.imgUrl]);
+
+	//////////////////
+	//   Functions  //
+	//////////////////
+
```

Comment:
Do you think the below functionality would work for retrieving Modal Content? I wasn't too sure how best to approach this from the BannerContent parent.